src/BaGet.Core/Storage/PackageStorageService.cs

```diff
@@ -108,6 +111,29 @@ public class PackageStorageService : IPackageStorageService
                 }
             }
 
+            // Store the package's icon, if one exists.
+            if (iconStream != null)
+            {
+                _logger.LogInformation(
+                    "Storing package {PackageId} {PackageVersion} icon at {Path}...",
+                    lowercasedId,
+                    lowercasedNormalizedVersion,
+                    iconPath);
+
+                result = await _storage.PutAsync(iconPath, iconStream, IconContentType, cancellationToken);
```

Comment:
Same comment as [here](https://github.com/loic-sharma/BaGet/pull/423/files#r368838123): let's try to detect the content type if possible